File: leetcode/509.c
int fib(int n){
    int fib2(int n, int a, int b) {
        if (n == 0) {
            return a;
        } else if ( n == 1 ) {
            return b;
        } else {
            return fib2(n-1, b, a + b);
        }
    }
  
    return fib2(n, 0, 1);
}

File: leetcode/509.kt
class Solution {
    fun fib(n: Int): Int {
        tailrec fun fib2(n: Int, a: Int, b: Int): Int = when (n) {
            0 -> a
            1 -> b
            else -> fib2(n - 1, b, a + b)
        }
        return fib2(n, 0, 1)
    }
}
